src/main/java/uk/gov/justice/hmpps/nomis/datacompliance/repository/jpa/repository/DuplicateOffenderRepository.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.justice.hmpps.nomis.datacompliance.repository.jpa.repository;
+
+
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+import uk.gov.justice.hmpps.nomis.datacompliance.repository.jpa.model.DuplicateOffender;
+
+import java.util.List;
+
+@Repository
+public interface DuplicateOffenderRepository extends CrudRepository<DuplicateOffender, String> {
+
+    /**
+     * The following query finds offenders that share the same
+     * PNC number as the reference offender. This check takes
+     * into account that either 2 or 4 digits may be used in
+     * the first part (year) of the number.
+     */
+    @Query(value =
+                    "SELECT O1.OFFENDER_ID_DISPLAY FROM OFFENDERS O1 " +
+                    "INNER JOIN OFFENDER_IDENTIFIERS OI1 " +
+                    "ON O1.OFFENDER_ID = OI1.OFFENDER_ID " +
+                    "AND OI1.IDENTIFIER_TYPE = 'PNC'" +
+                    "AND REGEXP_SUBSTR(UPPER(OI1.IDENTIFIER), '[0-9]{2,2}/[0-9]+[A-Z]+') IN (" +
+                    "    SELECT REGEXP_SUBSTR(UPPER(OI2.IDENTIFIER), '[0-9]{2,2}/[0-9]+[A-Z]+') FROM OFFENDERS O2" +
```

Comment:
Not sure what the regex does here, but I guess it copes with `2020/001234X` matching `2020/001234x` does it cope with 
`2019/001234X` matching `19/001234X` or 
`2019/001234X` matching `2019/1234X` or
`19/001234X` matching `2019/1234x` ?

Reply:
Though not sure how NOMIS pnc numbers are constrained so not sure what variations are in NOMIS.

Reply:
I just saw test, so it is the zero padding that might need a test assuming the regex works?

Reply:
It wasn't coping with zero padding, I'll add that in, thanks for flagging.